### PR TITLE
tests: Use proper temp file for scp pull testing

### LINKED ISSUE #782: Test suite leaves a `bar` test behind
Whenever I run the test suite, an empty `bar` file gets created in `$PWD`.

### PR DESCRIPTION
**Pull request by townsend2010**
Fixes #782

### CODE REVIEW

File: tests/test_scp_client.cpp [hunk 1]
```diff
@@ -123,16 +123,22 @@ TEST_F(SCPClient, throws_on_push_file_invalid_source)
 
 TEST_F(SCPClient, throws_when_pull_file_request_not_newfile)
 {
+    mpt::TempDir temp_dir;
+    auto file_name = temp_dir.path() + "/test-file";
```
Comment (ricab) [inline tests/test_scp_client.cpp]: Does the scp client handle the forward slash on windows?

Comment (ricab) [reply tests/test_scp_client.cpp]: I suppose so, as I see this in the code... https://github.com/CanonicalLtd/multipass/blob/4a9707473e1104ecb2d406fb37b0ebb469cfdbee/src/ssh/scp_client.cpp#L55

Comment (townsend2010) [reply tests/test_scp_client.cpp]: @ricab,

Yes, it's fine on Windows as other tests already use forward slashes and it works without issue.

Comment (ricab) [reply tests/test_scp_client.cpp]: ok, approving then

**Pull request merged**